types.go

```diff
@@ -121,6 +121,15 @@ type functionInfo struct {
 	statuser         Statuser
 }
 
+func (f functionInfo) canEvaluate(grad, hess bool) error {
+	if grad && !(f.IsGradient || f.IsFunctionGradient) {
+		return fmt.Errorf("function does not implement Grad() nor FuncGrad()")
```

Comment:
I see. I was just (wrongly) wondering if that "not" somehow implicitly substitutes "neither". Thanks for the explanation.